app/components/userSummary.tsx

```diff
@@ -146,14 +157,22 @@ const UserSummary: React.FC<UserSummaryProps> = ({ assignment, selectedUser, pro
 						)}
 					</div>
 				)}
-				{(viewer?.id === assignment.assignedUser?.id || !assignment.assignedUser) && <div className="sm:flex hidden items-start justify-center">
+				{(viewer?.id === assignment.assignedUser?.id && assignment.canBeDeleted === false) && <div className="sm:flex hidden items-start justify-center">
```

Comment:
Also I think we need to use same logic of showing deleting btns with viewer Id. It`s strange if other users can delete your assignment, what do u think?